src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -223,15 +224,16 @@ public void usedItem(DrinkConsumedEvent event, EntityRef item) {
      */
     @ReceiveEvent(components = {ThirstComponent.class})
     public void characterMoved(CharacterMoveInputEvent event, EntityRef character, ThirstComponent thirst) {
-        final float expectedDecay = event.isRunning() ? thirst.sprintDecayPerSecond : thirst.normalDecayPerSecond;
+        expectedDecay = event.isRunning() ? thirst.sprintDecayPerSecond : thirst.normalDecayPerSecond;
+        // Send event to allow for other systems to modify thirst decay.
+        AffectThirstEvent affectThirstEvent = new AffectThirstEvent(expectedDecay);
```

Comment:
While this "fixes" your problem that `AffectThirstEvent` is only called once (or when changing the movement mode) 

I'm wondering why it is **not called** when the periodic action is resolved? That would also solve your problem, I guess 🤔 At least, it should be consistent to _always_ send out `AffectThirstEvent` whenever we calculate and set the values on the thirst component.

Reply:
I kind of don't understand where that is taking place.... For hunger, it takes place along with hunger damage which has a period of 3 seconds. But, the thirst damage has a period of 60 seconds, so that can't be the only place where thirst is being updated, because I can see it updating every second or so.....
Changing the values of the thirst component automatically resulted in a change in the thirst decay... Now, the thirst decay occurs every second(or less) even while the player is stationary, just the decay per second changes value every time the player moves(if needed).

Reply:
I'm wondering whether it would make sense to align this with `Hunger` by reducing the period of the thirst damage periodic action and only checking the thirst then 🤔 I'm absolutely not sure, whether it's currently possible in-game to really run (without adjusting the speed via `setSpeedMultiplier`)... So _maybe_ we just get rid of that differentiation and (re-)introduce it for both Hunger and Thirst, once we actually _can_ have different movement modes in-game 🤷‍♀️

Reply:
ehm - I'm puzzled by how this actually works at all 🧐  You are right, the periodic action triggers only once per minute (I guess to have a slow decay when just idling). 

I fear fixing this turns out to be more elaborate work than initially planned 🙄 I'm tending towards the steps proposed by @jdrueckert:

- don't differentiate between running and walking (likely without effect anyways)
- reduce the interval of the periodic event (align with Hunger)
- only update the thirst value in a single place (taking elapsed time into account)
- add proper documentation for both system

Reply:
I would like to point out that the thirst actually decays in a very small interval (sub second), If you test out this PR and set the thirst multiplier in HyperthermiaSystem to 10 - 15, you can clearly see thirst decaying on a sub-second interval with a high rate. But the method that deals the thirst damage is not the one causing the decay since it has a period of 60 seconds. So there must be some other system which is causing the thirst decay. (thirst aurthority system probably only changes the thirst component of the entity and then some other system does the actual decaying part). However, I still do not get which is this other system.
Following this another thing popped into my mind: what if Hunger also changes in a similar way: even if the period of the hunger damage is 3 seconds, there could be some other system entirely which is decaying hunger in a sub-second interval..
And about isRunning(), won't it be easier to just to fix isRunning()?, I mean we can toggle walking/running in game, so we could simply change the value of  _running_ (the bool variable which is returned by isRunning) either by sending out an event or fixing whatever is in-place at the moment and not working...